src/api/c/unary.cpp

```diff
@@ -669,10 +669,12 @@ static af_err af_check(af_array *out, const af_array in) {
 
         // Convert all inputs to floats / doubles / complex
         af_dtype type = implicit(in_type, f32);
+        if(in_type == f16) type = f16;
```

Comment:
In this case, you can just call `af_dtype type = implicit(in_type, f16);`

Reply:
The implicit function will be used to convert integer types to floating point types. If we use f16 on the rhs, the integer types will be converted to f16 which has a pretty limited range compared to float.

Reply:
I have checked that, Integer types are not supported in this function

Reply:
We will probably never incude integer types for these three functions. Hence my comment